analyzers/src/SonarAnalyzer.CSharp/Rules/PropertiesAccessCorrectField.cs

```diff
@@ -119,6 +121,18 @@ protected override bool ShouldIgnoreAccessor(IMethodSymbol accessorMethod)
             || (property.GetMethod?.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax() is AccessorDeclarationSyntax getter
                 && getter.DescendantNodes().Any());
 
+        private static bool ContainsGetOrSetOnDependencyProperty(SyntaxNode syntaxNode, Compilation compilation)
+        {
+            var semanticModel = compilation.GetSemanticModel(syntaxNode.SyntaxTree);
```

Comment:
does this have a cost? could you first retrieve the `GetValue`/`SetValue` syntax nodes, and just after get, if the result is non-empty, get the semantic model for the compilation and do the check?

This should also separate the AST-only logic from the semantic logic. WDYT?

Reply:
I was curios about this and I've run the profiler:
![image](https://user-images.githubusercontent.com/56015273/116237732-e2e09480-a760-11eb-892e-99a3ebebef78.png)

As far as I can tell `GetSemanticModel` is not time consuming at all. Almost all execution time for this function is spent while running `DerivesFrom` (18ms from 18ms).

I would prefer to let this as it is since the code is easy to read and maintain.

Reply:
`IsMatch` looks like it could be improved but since it's just 1.33% from total rule execution time (18ms from 1346ms) I think that currently it doesn't worth the effort.